ureg/src/lib.rs

```diff
@@ -32,24 +32,28 @@ mod private {
 
 impl Uint for u8 {
     const TYPE: UintType = UintType::U8;
+    fn from_u32(val: u32) -> Self {
+        (val & 0xff) as u8
+    }
```

Comment:
Is this needed for multiple types? I think I would prefer to the the truncation closer to the specific register that needs it so that we can review specific usages as not dropping any meaningful bytes.

Reply:
It's not truly *needed* except on paper, as we don't have any 8- or 16-bit registers.

Reply:
I think we can drop this and replace it with `TryFrom<u32>`.

Reply:
I cleaned this up.